diff --git a/modules/custom/d_media/src/Plugin/Provider/ProviderPluginBase.php b/modules/custom/d_media/src/Plugin/Provider/ProviderPluginBase.php
index ad96eeb..ed49583 100644
--- a/modules/custom/d_media/src/Plugin/Provider/ProviderPluginBase.php
+++ b/modules/custom/d_media/src/Plugin/Provider/ProviderPluginBase.php
@@ -197,7 +197,7 @@ protected function calculateAspectRatio() {
   protected function getSpacerAttributes(array &$output) {
     $imageStyleSetting = $this->videoSettings['image_style'];

-    $effects = \Drupal::service('entity.manager')
+    $effects = \Drupal::service('entity_type.manager')
       ->getStorage('image_style')
       ->load($imageStyleSetting)->getEffects()->getConfiguration();
     foreach ($effects as $effect) {
